### ISSUE TITLE
Issue with image max-width

### ISSUE THREAD

**Issue by mojimi** [Kozea/WeasyPrint#776]
All I'm trying to achieve here is a Bootstrap-like row/col behaviour where I have a table with equal width cells, and that images inside these cells stay inside the cells.

I've tried display : flex and display : table, but both seem to be ignoring the img's max-width style property.

I'm not sure what other way I could try to have the image fit it's parent in a table-like matter.

html : 

```
<div class="row">
	<div class="col">Célula 1</div>
	<div class="col">Célula 2</div>
	<div class="col">Célula 3</div>
</div>
<div class="row">
	<div class="col">Célula 4</div>
	<div class="col">Célula 5</div>
	<div class="col">Célula 6</div>
</div>
<div class="row">
	<div class="col">Título 1</div>
</div>
<div class="row">
	<div class="col">
		<img src="someHugeImg">
	</div>
	<div class="col">
		<img src="someHugeImg">
	</div>
</div>
```
css : 

```
.row{
  text-align:center;
  width: 100%;
  display: table;
	border: 1px solid black;
}
.row > .col {
	display: table-cell;
}
img{
	width: 100%;
}
```
Working as expected in JsFiddle : http://jsfiddle.net/ppgab/zm7Lahdj/8/

**Comment by liZe** [comment]
Hello!

I'm not sure to understand what you want, as I get the same result in WeasyPrint and in my browser.

Here's what I get in WeasyPrint:
![capture d ecran de 2019-01-25 10-11-41](https://user-images.githubusercontent.com/952243/51736171-ae656600-2089-11e9-8a92-33dd104ec246.png)

And here's what I get in JsFiddle:
![capture d ecran de 2019-01-25 10-12-33](https://user-images.githubusercontent.com/952243/51736200-c3da9000-2089-11e9-8f4a-105c51f6da62.png)

Is that what you want?

**Comment by Tontyna** [comment]
@mojimi what's your WeasyPrint version?

**Comment by mojimi** [comment]
@liZe YES! That's exactly what I'm trying to achieve! 😞  But the images just stay at their original size...

I'm using the latest version that comes with PIP, version 44 @Tontyna.

I'm on a Windows 10 64 bits using Python 3.6.8, no errors are shown in log.

**Comment by liZe** [comment]
> That's exactly what I'm trying to achieve!

Well… That's already what I get with WeasyPrint 44 using the HTML and CSS from your JsFiddle :wink:.

**Comment by mojimi** [comment]
> 
> 
> > That's exactly what I'm trying to achieve!
> 
> Well… That's already what I get with WeasyPrint 44 using the HTML and CSS from your JsFiddle 😉.

Aw man... Do you know anything I could try? What could it be? There has to be something different...

**Comment by Tontyna** [comment]
> That's exactly what I'm trying to achieve!

... and what I get with WeasyPrint 44 on Windows 7

@liZe Is the downscaling delegated to Cairo? 

If yes: @mojimi What's your Cairo version?

**Comment by mojimi** [comment]
@Tontyna  the cairosvg python package is 2.2.1, the .dll in GTK3 i'm not sure how to check the version.

I just installed everything following the tutorial at the readthedocs documentation

**Comment by Tontyna** [comment]
Python snippet to get the Cairo version:

```python
import cairocffi as cairo
print('Cairo:', cairo.cairo_version_string ())
```

**Comment by mojimi** [comment]
> 
> 
> Python snippet to get the Cairo version:
> 
> ```python
> import cairocffi as cairo
> print('Cairo:', cairo.cairo_version_string ())
> ```

Version 1.16.0

Thanks for the quick replies

**Comment by Tontyna** [comment]
Same Cairo here. Hmm...

@mojimi Maybe a stupid question, but: Does the jsFiddle indeed fail for you? 
I assume the HTML / CSS you provided were reduced from your original, more comprehensive source, maybe leaving off the rule that prevents the 100% width...

**Comment by mojimi** [comment]
> 
> 
> Same Cairo here. Hmm...
> 
> @mojimi Maybe a stupid question, but: Does the jsFiddle indeed fail for you?
> I assume the HTML / CSS you provided were reduced from your original, more comprehensive source, maybe leaving off the rule that prevents the 100% width...

No, the jsFiddle works for me! It shows what I'm trying to achieve but the output from weasyprint doesn't.

And the HTML/CSS I provided is exactly the same as my original as I'm just testing for now 😬 .

But... I did some reinstalling and some cleaning and it works now! But using Bootstrap's flex row/columns still doesn't work, but that's fine, also if I'm not mistaken it does mention in the docs that flex is not perfect.

Thanks for the help! This library is amazing much better than the alternatives

**Comment by Tontyna** [comment]
> No, the jsFiddle works for me!

Wrong wordings in my stupid question -- of course I wanted to ask whether the example you provided fails in WeasyPrint...

...since it works now everywhere: :tada:

**Comment by liZe** [comment]
> Thanks for the help! This library is amazing much better than the alternatives

:smile:

> ...since it works now everywhere:

:tada:

**Comment by mojimi** [comment]
@liZe You can keep it closed, but I found out what the issue was.

In my .css example I accidentaly used width instead of just max-width.

By changing to max-width the problem does comeback, but just using width is fine for me.

**Comment by liZe** [comment]
It's probably related to #656.
